crypto/fipsmodule/rsa/rsa_impl.c

```diff
@@ -1352,10 +1356,10 @@ static int RSA_generate_key_ex_maybe_fips(RSA *rsa, int bits,
 
     // Only retry on |RSA_R_TOO_MANY_ITERATIONS|. This is so a caller-induced
     // failure in |BN_GENCB_call| is still fatal.
-  } while (failures < 4 && ERR_GET_LIB(err) == ERR_LIB_RSA &&
+  } while (failures < MAX_KEYGEN_ATTEMPTS && ERR_GET_LIB(err) == ERR_LIB_RSA &&
```

Comment:
This will always evaluate to false in non-fips mode:

If `rsa_generate_key_impl` succeeds then we break out of the do-while loop

If `rsa_generate_key_impl` fails we increment `failures` from 0 to 1. But since `MAX_KEYGEN_ATTEMPTS = 1` the condition will evaluate to false and we proceed without a retry.

This is prob screwing with the test fixture `KeygenInternalRetry`.